variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -51,45 +54,66 @@
      * @param line Contents of the line in the file
      * @return The list of Variant objects that can be created using the fields from a VCF record
      */
-    public List<Variant> create(String fileId, String studyId, String line) throws IllegalArgumentException {
+    public List<Variant> create(String fileId, String studyId, String line)
+            throws IllegalArgumentException, NonVariantException, IncompleteInformationException {
+
         String[] fields = line.split("\t");
         if (fields.length < 8) {
             throw new IllegalArgumentException("Not enough fields provided (min 8)");
         }
 
-        String chromosome = getChromosomeWithoutPrefix(fields);
-        int position = getPosition(fields);
-        Set<String> ids = new HashSet<>(); //EVA-942 - Ignore IDs submitted through VCF
-        String reference = getReference(fields);
-        String[] alternateAlleles = getAlternateAlleles(fields, chromosome, position, reference);
+        String[] alternateAlleles = getAlternateAlleles(fields);
+        List<VariantCoreFields> generatedKeyFields = buildVariantCoreFields(fields, alternateAlleles);
```

Comment:
I find the previous implementation more consistent. I understand that the alternates can't be calculated inside `buildVariantCoreFields`, but it looks quite strange that fields are provided in different ways.